pkg/module/parse/template_to_module.go

```diff
@@ -150,13 +154,22 @@ func (p *Parser) newManifestFromTemplate(
 		if err != nil {
 			return nil, err
 		}
-		componentDescriptor, err = p.ComponentDescriptorCache.GetRemoteDescriptor(ctx,
-			descriptorCacheKey, descriptor, p.Client)
+		componentDescriptor, err = parser.ComponentDescriptorCache.GetRemoteDescriptor(ctx,
+			descriptorCacheKey, descriptor, parser.Client)
 		if err != nil {
 			return nil, err
 		}
 	}
 
+	verification, err := signature.NewVerification(ctx,
+		parser.Client,
+		parser.EnableVerification,
+		parser.PublicKeyFilePath,
+		module.Name)
+	if err != nil {
+		return nil, err
+	}
+
 	if err := signature.Verify(componentDescriptor, verification); err != nil {
 		return nil, fmt.Errorf("could not verify descriptor: %w", err)
```

Comment:
Better to specify the message log as it's a signature verification than a general template verification. Maybe: "could not verify signature"